### PR TITLE
Add all meaningful augmentations from TorchIO

### LINKED ISSUE #248: Squeeze more juice out of TorchIO augmentation

### PR DESCRIPTION
**Pull request by dzenanz**
Closes #248.

### CODE REVIEW

File: miqa/learning/nn_training.py [hunk 1]
```diff
@@ -242,6 +243,97 @@ def apply_transform(self, subject: torchio.Subject) -> torchio.Subject:
...
+        if original_quality < 6 and len(subject.applied_transforms) == 1:  # low quality image
+            return subject
+        else:  # high-quality image, corrupt it
+            transformed_subject = super().apply_transform(subject)
+
+            # now determine how much quality was reduced
+            # applied_params = transformed_subject.applied_transforms[-1][1]
+            # coefficients = applied_params['coefficients']['img']
+            # quality_reduction = 2 + np.linalg.norm(np.asarray(coefficients))
+            quality_reduction = 3  # it is hard to assess impact on image quality
```
Comment (annehaley) [inline miqa/learning/nn_training.py:258]: Is this magic number 3 temporary? If not, what helped you reach this number? I noticed in the notebook that the quality reduction was coming out to about 2.2.

**Review by annehaley — COMMENTED** [review]
I recommend updating the title of this PR to make its purpose clearer. The images output from the ipynb look nice!

File: miqa/learning/nn_training.py (hunk 1 above)
Comment (dzenanz) [reply miqa/learning/nn_training.py:258]: This number is my initial guess of which number would work the best for affecting the learning in such a way to maximize our accuracy on validation data. I came up with it by looking at effect of different bias fields, and my inability to come up with a good formula which transforms those coefficients into quality loss. I also compared the biased images from notebook with [PredictHD description](https://docs.google.com/presentation/d/1aw9VbUvBYz_NuCuahrUNiqPyU3k1yzuk/edit?usp=sharing&ouid=118307775775031133917&rtpof=true&sd=true) and images therein, namely slides 38-43.

This number should correlate the quality reduction of the bias field which we just produced, as compared to reference training data.

I intend to experiment with this number, to see what works the best (gives highest accuracy).

**Comment by dzenanz** [comment]
After I created the PR, I realized that the title was picked up from the first, 1-line commit 😖

I guess I changed the title while you were reviewing 😄

File: miqa/learning/nn_training.py (hunk 1 above)
Comment (annehaley) [reply miqa/learning/nn_training.py:258]: Okay, I figured you intended to change this later. Sounds good to me.

**Comment by annehaley** [comment]
> I guess I changed the title while you were reviewing

That's exactly what happened, whoops! All's well that ends well :grin:

**Review by annehaley — APPROVED** [review]
